backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/EakApplicationMutationService.kt

```diff
@@ -128,4 +131,31 @@ class EakApplicationMutationService {
             }
         }
     }
+
+    @GraphQLDescription("Updates a note of an application")
+    fun updateApplicationNote(
+        applicationId: Int,
+        noteText: String,
+        dfe: DataFetchingEnvironment
+    ): Boolean {
+        val context = dfe.getContext<GraphQLContext>()
+        val jwtPayload = context.enforceSignedIn()
+
+        return transaction {
+            val application = ApplicationEntity.findById(applicationId) ?: throw UnauthorizedException()
```

Comment:
Unauthorized ist not the correct exception here, the message of UnauthorizedExeption ist ""JWT expired, invalid or missing".

Definition for 401 is: "JWT expired, invalid or missing".

You should either use 403 Forbidden or (what I think is more appropriate here) a 404 (Not found). I think 404 is the better choice here, as we do not provide information to someone unauthorized to know if this resource exists or not.

Reply:
Alright makes sense
Just a question to @michael-markl 
You've added this comment
```
val application = ApplicationEntity.findById(applicationId) ?: throw UnauthorizedException()
            // We throw an UnauthorizedException here, as we do not know whether there was an application with id
            // `applicationId` and whether this application was contained in the user's project & region.

```
Does this really make sense? I mean you can not delete an application if you were not able to display it. Sure you could inject any id you want to but in `deleteApplication` a NotFoundExceptions makes also more sense?!

Reply:
I am also partially confused with this comment :D I agree with @ztefanie that either 403 or 404 seems correct, maybe 404 is preferred as it's clearer what happend, 403 might be in some scenario potentially safer (to prevent scraping of active application ids crossproject/crossregion) but at the end it does not really matter probably :D